re.c

```diff
@@ -2954,7 +2954,7 @@ static void
 reg_resize(regex_t *reg)
 {
     if (reg->alloc > reg->used) {
-      unsigned char *new_ptr = xrealloc(reg->p, reg->used);
+      unsigned char *new_ptr = ruby_sized_xrealloc(reg->p, reg->used, reg->capa);
```

Comment:
Do not use `ruby_sized_xrealloc` in this file.
`reg->p` is allocated in `onig_bbuf_init` where `xmalloc` is not `ruby_xmalloc`.